src/frameworks/vite/index.ts

```diff
@@ -68,20 +65,21 @@ export async function build(root: string) {
 
   await warnIfCustomBuildScript(root, name, DEFAULT_BUILD_SCRIPT);
 
-  await build({ root });
+  await build({ root, mode: "production" });
 }
 
 export async function ɵcodegenPublicDirectory(root: string, dest: string) {
   const viteConfig = await getConfig(root);
-  const viteDistPath = join(root, viteConfig.build.outDir);
+  const viteDistPath = join(root, viteConfig!.build.outDir);
```

Comment:
shouldn't this handle undefined `viteConfig`?

```ts
  const viteConfig = await getConfig(root);
  if(!viteConfig) ..

  const viteDistPath = join(root, viteConfig.build.outDir);
```

Reply:
We fail earlier in the build process, if discover fails, see the throw in frameworks/index.ts but might be worth being defensive

Reply:
I found the issues with sveltekit that i was guarding here, will ditch the try/catch